### PR TITLE
Add preview for Multi-Currency single currency settings page.

### LINKED ISSUE #1928: Preview on single currency setting should show the preview amount
Within an enabled currency under WooCommerce > Settings > Multi-Currency there is a preview to show what the amount will look like after the conversion, charm, and rounding are applied. This needs to be made functional.

Screen Shot 2021-05-27 at 1 54 54 PM

### PR DESCRIPTION
**Pull request by jessepearson**
Fixes #1928

#### Changes proposed in this Pull Request

* Add hidden field on single currency settings page to hold automatic rate value.
* Clean up available values for rounding and charm rates, make charm select instead of text.
* Add preview fields, and make them functional.
* Set manual rate to default to automatic rate if empty or below zero.

#### Testing instructions

* Navigate to WooCommerce > Settings > Multi-Currency
* If no additional currencies are enabled, use the Add Currencies button to add one.
* Click on the pencil to edit the currency.
* Update values on the settings page to trigger the preview field to update.

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by LevinMedia** [comment]
Working as expected! However, the preview field here should not be editable. 


Screen Shot 2021-06-08 at 1 46 58 PM

If it's possible, I'd prefer the preview output not appear in a field at all, and instead sit on the background of the page, in a fashion more similar to the original design, however if this is a constraint of the legacy settings screen we can keep it as is, minus the edit-ability. 

7KVvtMQgJxVAUcPPxTB29r-fi-241%3A888

**Comment by jessepearson** [comment]
@LevinMedia I made some updates I believe you will be happy with 😁 
Screen Capture on 2021-06-09 at 09-27-00

**Comment by ismaeldcom** [comment]
Hey @jessepearson! I created 2 issues, and realize now that they are partially solved by this PR.

The only missing thing is the price rounding help tooltip mention dollar in `Rounding up to the nearest whole dollar helps prevent losses on sales.` It should use the edited currency or a more general word like `whole number` or `integer`. What do you think?

The free shipping having price is solved now with this PR not allowing positive charming 👍

**Comment by jessepearson** [comment]
Hey @ismaeldcom , good catch on that. Updated in https://github.com/Automattic/woocommerce-payments/pull/2064/commits/a4d4a2f394110e39ba063db74d793bb2a70efb94

**Review by ismaeldcom — APPROVED** [review]
LGTM! I like the extra care for non valid numbers 😄 

I left a little nitpick comment, but feel free to 🚢 it whenever you want!

File: client/multi-currency/index.js [hunk 1]
```diff
@@ -35,3 +82,19 @@ document.querySelectorAll( '.exchange-rate-selector' ).forEach( ( radio ) => {
...
+	updatePreview();
+} );
+
+charm.addEventListener( 'input', () => {
+	updatePreview();
+} );
+
+previewAmount.addEventListener( 'input', () => {
+	updatePreview();
+} );
```
[the remark this answers is not in the document]
Comment (jessepearson) [reply client/multi-currency/index.js]: I knew there was probably a more concise way to do this. Thanks for the suggestion, I just added it 😀

**Pull request merged**